test-e2e/test/helpers.js

```diff
@@ -0,0 +1,34 @@
+const path = require('path');
+const fs = require('fs-extra');
+const childProcess = require('child_process');
+const {promisify} = require('util');
+
+const ESYCOMMAND = require.resolve('../../bin/esy');
+
+const promiseExec = promisify(childProcess.exec);
+
+const esyCommands = {
```

Comment:
I'm not sure about this. It feels nice to use it that way as it's less boilerplate and there is less to keep track of.

Something like this could however be nice to expose as a node lib.